bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/discovery/SiemensHvacDeviceDiscoveryService.java

```diff
@@ -0,0 +1,182 @@
+
+    public SiemensHvacDeviceDiscoveryService() {
+        super(SUPPORTED_THING_TYPES, SEARCH_TIME);
+    }
+
+    @Reference
+    public void setSiemensHvacMetadataRegistry(@Nullable SiemensHvacMetadataRegistry metadataRegistry) {
+        this.metadataRegistry = metadataRegistry;
+    }
+
+    public void unsetSiemensHvacMetadataRegistry(SiemensHvacMetadataRegistry metadataRegistry) {
+        this.metadataRegistry = null;
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        // can be overridden
+    }
+
+    private @Nullable ThingUID getThingUID(ThingTypeUID thingTypeUID, String serial) {
+        final SiemensHvacBridgeBaseThingHandler lcSiemensHvacBridgeHandler = siemensHvacBridgeHandler;
+        if (lcSiemensHvacBridgeHandler != null) {
+            ThingUID localBridgeUID = lcSiemensHvacBridgeHandler.getThing().getUID();
+            return new ThingUID(thingTypeUID, localBridgeUID, serial);
+        }
+        return null;
+    }
+
+    @Override
+    public void startScan() {
+        final SiemensHvacMetadataRegistry lcMetadataRegistry = metadataRegistry;
+        final SiemensHvacBridgeBaseThingHandler lcSiemensHvacBridgeHandler = siemensHvacBridgeHandler;
+        logger.debug("call startScan()");
+
+        if (lcMetadataRegistry != null) {
+            lcMetadataRegistry.ReadMeta();
+
+            // SiemensHvacMetadataMenu rootMenu = metadataRegistry.getRoot();
+            // for (SiemensHvacMetadata child : rootMenu.getChilds().values()) {
+            ArrayList<SiemensHvacMetadataDevice> devices = lcMetadataRegistry.getDevices();
+
+            if (devices == null) {
+                return;
+            }
+
+            for (SiemensHvacMetadataDevice device : devices) {
+
+                String name = device.getName();
+                String type = device.getType();
+                String addr = device.getAddr();
+                String serialNr = device.getSerialNr();
+
+                if (name.indexOf("OZW672") >= 0) {
+                    continue;
+                }
+
+                logger.debug("Find devices: {} / {} / {} / {}", name, type, addr, serialNr);
+
+                String typeSn = UidUtils.sanetizeId(type);
+                ThingTypeUID thingTypeUID = new ThingTypeUID(SiemensHvacBindingConstants.BINDING_ID, typeSn);
+
+                ThingUID thingUID = getThingUID(thingTypeUID, serialNr);
+
+                if (lcSiemensHvacBridgeHandler != null) {
+                    ThingUID bridgeUID = lcSiemensHvacBridgeHandler.getThing().getUID();
+
+                    if (thingUID != null) {
+                        Map<String, Object> properties = new HashMap<>(1);
+                        properties.put("name", name);
+                        properties.put("type", type);
+                        properties.put("addr", addr);
+                        properties.put("serialNr", serialNr);
```

Comment:
What if the device is created by a things file ? Are the properties also populated ?